src/ast/nodes/IfStatement.ts

```diff
@@ -124,11 +124,12 @@ export default class IfStatement extends StatementBase implements DeoptimizableE
 
 	private getTestValue(): LiteralValueOrUnknown {
 		if (this.testValue === unset) {
-			return (this.testValue = this.test.getLiteralValueAtPath(
-				EMPTY_PATH,
-				SHARED_RECURSION_TRACKER,
-				this
-			));
+			const testValue = this.test.getLiteralValueAtPath(EMPTY_PATH, SHARED_RECURSION_TRACKER, this);
+			if (testValue === UnknownFalsyValue) {
+				this.testValue = false;
+			} else {
+				this.testValue = testValue;
+			}
```

Comment:
Nice idea. Why not take it to the next level by writing a general helper method `tryCastLiteralValueToBoolean` that maps `UnknownFalsyValue` to `false`, but also `UnknownTruthyValue` to `true` and other symbol values to `UnknownValue`? And then we use the method in three places:

1. here
```suggestion
  		const testValue = tryCastLiteralValueToBoolean(getLiteralValueAtPath(EMPTY_PATH, SHARED_RECURSION_TRACKER, this));
```
2. In ConditionalExpression
3. In LogicalExpression

This will also ensure that objects are "truthy". And having a central helper for all places where we test a literal value for its boolean value ensures that the logic can be easily improved in a central place.

Reply:
Hahaha, I was actually planning to update `ConditionalExpression` and `LogicalExpression ` in a follow-up PR.

Yeah, introducing `tryCastLiteralValueToBoolean` method is definitely better, especially since `ConditionalExpression` and `LogicalExpression` will also use this method.

I'll make the updates as soon as possible.